testing/citest/tests/kube_smoke_test.py

```diff
@@ -235,6 +234,159 @@ def delete_load_balancer(self):
+            status_class=st.SynchronousHttpOperationStatus),
+        contract=builder.build())
+
+  def save_statefulset_pipeline(self):
+      bindings = self.bindings
+      payload = self.agent.make_json_payload_from_kwargs(
+			    job=[{
+	                'keepWaitingPipelines': 'false',
+                        'application': self.TEST_APP,
+                        'name': 'statefulset-pipeline',
+	                'lastModifiedBy': 'anonymous',
+	                'limitConcurrent': 'true',
+	                'parallel': 'true',
+	                'stages': [{
+		                'clusters': [{
+			                'account': bindings['SPINNAKER_KUBERNETES_ACCOUNT'],
+			                'application': self.TEST_APP,
+			                'cloudProvider': 'kubernetes',
+			                'containers': [{
+				                    'imageDescription': {
+					                'repository': 'nginx'
+				                        },
+				            'imagePullPolicy': 'ALWAYS',
+				            'limits': {
+					            'memory': '200Mi'
+				                    },
+				            'name': 'nginx',
+				            'requests': {
+					            'cpu': '100m',
+					            'memory': '200Mi'
+				                }
+			                }],
+			        'kind': 'StatefulSet',
+			        'namespace': 'default',
+			        'provider': 'kubernetes',
+			        'region': 'default',
+			        'targetSize': 0,
+			        'terminationGracePeriodSeconds': 30
+		                }],
+		                'name': 'Deploy',
+		                'refId': '1',
+
+		            'type': 'deploy'
+	                }],
+	                'triggers': [{'enabled': 'true'}]
+                }],
+                    description='Save Statefulsets:' ,application=self.TEST_APP)
+      builder = kube.KubeContractBuilder(self.kube_observer)
+      builder.new_clause_builder('Run pipeline', retryable_for_secs=15).get_resources('svc',extra_args=['--namespace', self.TEST_NAMESPACE],no_resource_ok=True)
+      my_dict = eval(payload)
+      val=my_dict['job']
+      val=str(val[0])
+      val=val.replace('\'','\"')
+      return st.OperationContract(self.new_post_operation(title='create_state_ful_sets', data=val, path='pipelines',status_class=st.SynchronousHttpOperationStatus),contract=builder.build())
+
+  def create_daemonset_pipeline(self):
+    bindings = self.bindings
+    payload = self.agent.make_json_payload_from_kwargs(
+        job=[{
+            'type': 'manual',
+            'user': '[anonymous]'
+        }],
+        description='Create Daemonset'.format(
+            self.__lb_name,
+            bindings['SPINNAKER_KUBERNETES_ACCOUNT']),
+        application=self.TEST_APP)
+    builder = kube.KubeContractBuilder(self.kube_observer)
+    (builder.new_clause_builder('Run pipeline', retryable_for_secs=15)
+     .get_resources(
+         'svc',
+         extra_args=['--namespace', self.TEST_NAMESPACE],
+         no_resource_ok=True))
```

Comment:
similar to above, I dont think you are checking for anything here. You allow it to be empty but do not require it to be empty.